Source/WebCore/Modules/mediastream/gstreamer/GStreamerMediaEndpoint.cpp

```diff
@@ -731,24 +750,17 @@ void GStreamerMediaEndpoint::addRemoteStream(GstPad* pad)
 
     unsigned mLineIndex;
     g_object_get(rtcTransceiver.get(), "mlineindex", &mLineIndex, nullptr);
-    const auto* media = gst_sdp_message_get_media(description->sdp, mLineIndex);
-    GUniquePtr<char> sdp(gst_sdp_media_as_text(media));
-    auto sdpString = makeString(sdp.get());
 
+    // Look-up the mediastream ID, using the msid attribute, fall back to pad name if there is no msid.
+    const auto* media = gst_sdp_message_get_media(description->sdp, mLineIndex);
     GUniquePtr<gchar> name(gst_pad_get_name(pad));
-    auto label = String::fromLatin1(name.get());
-    auto key = makeString("msid:");
-    auto lines = sdpString.split('\n');
-    for (auto& line : lines) {
-        auto i = line.find(key);
-        if (i != notFound) {
-            auto tmp = line.substring(i + key.ascii().length());
-            label = tmp.left(tmp.find(' '));
-            break;
-        }
+    auto mediaStreamId = String::fromLatin1(name.get());
+    if (const char* msidAttribute = gst_sdp_media_get_attribute_val(media, "msid")) {
+        auto components = makeString(msidAttribute).split(' ');
```

Comment:
StringView?

Reply:
Way less idiomatic though, also the split returns an iterator of StringView, i need a String back, so I don't see a way around.